image_client/picturae_importer.py

```diff
@@ -391,6 +394,42 @@ def populate_taxon(self):
         else:
             pass
 
+    def generate_taxon_fields(self, index, taxon):
```

Comment:
split of part of the taxon_create funcition into a new funciton generaate_taxon_fields, to allow for easier modular testing, and a shorter function.